files/class-api-client.php

```diff
@@ -188,4 +188,21 @@ public function is_file( $file_path, &$info = null ) {
 		/* translators: 1: file path 2: HTTP status code */
 		return new WP_Error( 'is_file-failed', sprintf( __( 'Failed to check if file `%1$s` exists (response code: %2$d)' ), $file_path, $response_code ) );
 	}
+
+	public function get_unique_name( $file_path ) {
+		$response = $this->call_api( $file_path, 'get', [
+			'headers' => [
+				'x-action' => 'unique_filename',
+			],
+		] );
+
+		if ( is_wp_error( $response ) ) {
+			return $response;
+		}
+
+		$response_code = wp_remote_retrieve_response_code( $response );
+		$content = wp_remote_retrieve_body( $response );
+
+		return compact( 'response_code', 'content' );
```

Comment:
Rather than return the response code and content, we should probably think about how we can improve the return from this function based on how we expected to be used. For example, should we return an error if the filename is not unique?

We should also add some tests for this new functionality (there are tests for the Api_Client already, for example).

Reply:
That's a good point. I'll get started working on that.